polkadot/node/core/backing/src/lib.rs

```diff
@@ -232,8 +234,8 @@ struct PerRelayParentState {
 	inject_core_index: bool,
 	/// The core states for all cores.
 	cores: Vec<CoreState>,
-	/// The validator groups at this relay parent.
-	validator_groups: Vec<Vec<ValidatorIndex>>,
+	/// The validator index -> group mapping at this relay parent.
+	validator_to_group: IndexedVec<ValidatorIndex, Option<GroupIndex>>,
```

Comment:
Is it really necessary to copy this to the relay parent state? Can't we just keep the session index here and then fetch a reference from state directly where we need it?

Reply:
this being an LruMap, it makes it a bit harder to reason about the existence of the session index in the map.

And we can't store a mutable reference here and just populate the map when the session is not present because we'd need to hand out multiple mutable references.

Realistically, backing can only happen on the current session, right? So the LruMap doesn't need a capacity larger than 1 (so then, it doesn't even need to be a LruMap). Am I right?

Reply:
with current async backing code, we cannot start backing a candidate in the next session in advance. But we may want to keep this possibility, making the LruMap neccessary.

What I can do is wrap it in an Arc, because they don't need to modify it. Which would achieve the same

Reply:
done. I'll merge it once CI passes again